File: 6-ZigzagConversion/6-ZigzagConversion.py
# Last updated: 4/24/2026, 4:52:11 PM
1class Solution:
2    def convert(self, s: str, numRows: int) -> str:
3        if numRows == 1 or len(s) <= numRows: return s
4        rows = [""] * numRows
5        idx = 0
6        count = 0
7        down = False
8        for ch in s:
9            rows[idx] += ch
10            if idx == 0 or idx == numRows - 1:
11                down = not down
12            idx += 1 if down else -1
13            count += 1
14        return ''.join(rows)
15

File: 6-ZigzagConversion/6-ZigzagConversion.cpp
// Last updated: 4/25/2026, 8:21:31 PM
1class Solution {
2public:
3    string convert(string s, int numRows) {
4        if (numRows == 1 or s.size() <= numRows) return s;
5        vector<string> rows(numRows, "");
6        int rowIdx = 0;
7        bool down = false;
8        for (auto& i : s) {
9            rows.at(rowIdx) += i;
10            if (rowIdx == 0 or rowIdx == (numRows - 1)) down = not down;
11            rowIdx += down ? 1 : -1;
12        }
13        string ret;
14        for (auto& i : rows) ret += i;
15
16        return ret;
17    }
18};